Improved Neovim support and script to replace configs

Tehnix opened the pull request:
Add script to backup any existing neovim configuration, and symlink the HVN configuration
into ~/.config/nvim/.

Also adds some neovim specific settings into the .vimrc, wrapped in checks for
has('nvim'), so they don't affect normal vim.

This closes #87 and closes #114. Sorry for the duplicate pull request in #123.

Tehnix commented:
To see the earlier discussion, check out #123. Unfortunately I'm not such a git pull request ninja, so this was the quickest way I knew how to clean up the commits.

SimpleX91 commented:
Looks good to me. Could be nice to run plugins setup just like for vim.

begriffs commented on .vimrc:
```diff
@@ -655,13 +684,13 @@ let g:syntastic_haskell_hdevtools_args = '-g-Wall'
 nnoremap <silent> <leader>hh :Hoogle<CR>
 
 " Hoogle and prompt for input
-nnoremap <leader>hH :Hoogle 
```
It's important to preserve this final space character actually.

begriffs commented on .vimrc:
```diff
@@ -655,13 +684,13 @@ let g:syntastic_haskell_hdevtools_args = '-g-Wall'
 nnoremap <silent> <leader>hi :HoogleInfo<CR>
 
 " Hoogle for detailed documentation and prompt for input
-nnoremap <leader>hI :HoogleInfo 
```
This final space character should be preserved as well.

SimpleX91 commented:
@Tehnix to rebase *this* branch you need to:
1. make changes and commit;
2. run `git rebase --interactive master`;
3. follow instructions in comments (place your latest commit right after "Improve NeoVim support" and change `pick` to `fixup`), should be something like
```
pick 12345 Improve NeoVim ..
fixup 12346 Preserve hoogle spaces
pick 12347 Add script ..
```
4. check `git log`, there should be only two commits left and diff should not contain those spaces removal;
5. if all OK, run `git push --force`.

That's all, simple as that =) Btw, rebasing your changes is a good practice to keep commits log clean and pretty (no things like "fixed this typo" and "fixed it again, this time for sure").  You can do enything you want to your changes *before* PR gets merged.
Learn how to use rebase, it's really helpful and totally handy. ;)

Tehnix commented:
Added the spaces (I think my editor had removed them :) ), and rebased to current master.

@SimpleX91 regarding the plugin support, the `neovim.sh` script assumes that you have run the normal install script first. It would perhaps be nice to abstract some of the parts of `setup.sh` out, and then make `neovim.sh` standalone, so it doesn't matter which is run, and in what order - but I think that is for another pull request :)

This should be able to be merged now.

Tehnix commented:
Oh, and thanks for the rebase help, @SimpleX91 ! Very much appreciated, I'm so used to working alone in my git repos, that I'm very rusty when it comes to rebasing :)

begriffs commented:
Thank you! I tried in neovim locally and it works great. Lots of good changes in your PR.
